db/builder.cc

```diff
@@ -306,12 +324,17 @@ Status BuildTable(
                                        total_tombstone_payload_bytes;
         uint64_t total_payload_bytes_written =
             (tp.raw_key_size + tp.raw_value_size);
+        // TODO(yuzhangyu): update this to include number of range delete
+        // entries.
+        uint64_t stripped_ts_bytes =
+            strip_timestamp ? ts_sz * tp.num_entries : 0;
         // Prevent underflow, which may still happen at this point
         // since we only support inserts, deletes, and deleteRanges.
         if (total_payload_bytes_written <= total_payload_bytes) {
           *memtable_payload_bytes = total_payload_bytes;
           *memtable_garbage_bytes =
-              total_payload_bytes - total_payload_bytes_written;
+              total_payload_bytes -
+              (total_payload_bytes_written + stripped_ts_bytes);
```

Comment:
I don't think this change belongs here. The garbage measurement is intended to indicate how much would be saved by garbage collecting the memtable (currently mempurge) rather than flushing.

Reply:
I see, so these stats are for evaluating the effectiveness of the mempurge feature. This flag is not compatible with the feature yet, so probably don't need to update how these bytes are calculated when these two features are enabled together. Also as you suggested, these stats are not for evaluating the flushing process, so doesn't make much sense to interfere it with the stripped udt bytes.